cdc_vaccines/delphi_cdc_vaccines/pull.py

```diff
@@ -0,0 +1,162 @@
+# -*- coding: utf-8 -*-
+"""Functions for pulling data from the CDC data website for vaccines."""
+import hashlib
+from logging import Logger
+from delphi_utils.geomap import GeoMapper
+import numpy as np
+import pandas as pd
+
+
+
+
+
+def pull_cdcvacc_data(base_url: str, logger: Logger) -> pd.DataFrame:
+    """Pull the latest data from the CDC on vaccines and conform it into a dataset.
+
+    The output dataset has:
+    - Each row corresponds to (County, Date), denoted (FIPS, timestamp)
+    - Each row additionally has columns that correspond to the counts or
+      cumulative counts of vaccination status (fully vaccinated,
+      partially vaccinated) of various age groups (all, 12+, 18+, 65+)
+      from December 13th 2020 until the latest date
+
+    Note that the raw dataset gives the `cumulative` metrics, from which
+    we compute `counts` by taking first differences.  Hence, `counts`
+    may be negative.  This is wholly dependent on the quality of the raw
+    dataset.
+
+    We filter the data such that we only keep rows with valid FIPS, or "FIPS"
+    codes defined under the exceptions of the README.  The current  exceptions
+    include:
+    # - 0: statewise unallocated
```

Comment:
Sure I'll take a look on Monday!

Reply:
Alright, looking through our chats on prop signals, this is what can happen: if megacounties are used to keep track of unassigned counts, then we must be careful not to add the megacounty population counts when doing prop signals, otherwise when aggregating to larger geos we will double count the state population. I don't see any prop signals atm, so this seems fine.

Reply:
Okay! So from what I understand, I'll leave as in for now!